File: iOSDemoProject/iOSDemoProject/Bridges/RNBridge/RNRouterModule.swift
import Foundation

// MARK: - RCT Router Module (placeholder for actual React Native integration)
/// JavaScript usage:
/// ```js
/// import { NativeModules } from 'react-native';
/// NativeModules.AppRouterModule.navigate('app://flutter/home');
/// NativeModules.AppRouterModule.pop();
/// ```
class RNRouterModule {

    func navigate(_ url: String) {
        AppRouter.shared.navigate(to: url)
    }

    func pop() {
        AppRouter.shared.pop()
    }
}

File: AndroidDemoProject/app/src/main/java/com/by/androiddemoproject/bridges/rn/RNRouterModule.kt
package com.by.androiddemoproject.bridges.rn

import android.content.Context
import com.by.androiddemoproject.router.AppRouter

// MARK: - ReactContextBaseJavaModule placeholder for routing
// JavaScript usage:
// ```js
// import { NativeModules } from 'react-native';
// NativeModules.AppRouterModule.navigate('app://flutter/home');
// ```

class RNRouterModule {

    // @ReactMethod
    fun navigate(context: Context, url: String) {
        AppRouter.navigate(context, url)
    }
}
